temporal-sdk/src/main/java/io/temporal/internal/worker/LocalActivityWorker.java

```diff
@@ -100,67 +132,96 @@ public void start() {
+      return accepted;
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      return false;
+    }
   }
 
-  @Override
-  public void awaitTermination(long timeout, TimeUnit unit) {
-    poller.awaitTermination(timeout, unit);
+  /**
+   * @param executionContext execution context of the activity
+   * @param delayMs delay time in milliseconds to the next attempt
+   * @param failure if supplied, it will be used to override {@link
+   *     LocalActivityExecutionContext#getLastFailure()}
+   */
+  private void scheduleNextAttempt(
+      LocalActivityExecutionContext executionContext, long delayMs, @Nullable Failure failure) {
+    PollActivityTaskQueueResponse nextActivityTask =
+        executionContext.getNextAttemptActivityTask(failure);
+    LocalActivityAttemptTask task =
+        new LocalActivityAttemptTask(executionContext, nextActivityTask);
+    Deadline.after(delayMs, TimeUnit.MILLISECONDS)
+        .runOnExpiration(new LocalActivityRetryHandler(task), scheduledExecutor);
   }
 
-  @Override
-  public void suspendPolling() {
-    poller.suspendPolling();
-  }
+  private class LocalActivityDispatcherImpl implements LocalActivityDispatcher {
+    @Override
+    public boolean dispatch(
+        ExecuteLocalActivityParameters params,
+        Functions.Proc1<LocalActivityResult> resultCallback) {
+
+      long localRetryThresholdMs = params.getLocalRetryThreshold().toMillis();
+      Preconditions.checkState(localRetryThresholdMs > 0, "localRetryThresholdMs must be > 0");
+      Deadline localRetryDeadline = Deadline.after(localRetryThresholdMs, TimeUnit.MILLISECONDS);
+
+      long passedFromOriginalSchedulingMs = 0;
+      if (params.getOriginalScheduledTimestamp() != ExecuteLocalActivityParameters.NOT_SCHEDULED) {
+        passedFromOriginalSchedulingMs =
+            System.currentTimeMillis() - params.getOriginalScheduledTimestamp();
+      } else {
+        params.setOriginalScheduledTimestamp(System.currentTimeMillis());
+      }
 
-  @Override
-  public void resumePolling() {
-    poller.resumePolling();
-  }
+      Duration scheduleToCloseTimeout = params.getScheduleToCloseTimeout();
+      Deadline scheduleToCloseDeadline = null;
+      if (scheduleToCloseTimeout != null) {
+        scheduleToCloseDeadline =
+            Deadline.after(
+                scheduleToCloseTimeout.toMillis() - passedFromOriginalSchedulingMs,
+                TimeUnit.MILLISECONDS);
+      }
 
-  @Override
-  public boolean isSuspended() {
-    return poller.isSuspended();
-  }
+      LocalActivityExecutionContext executionContext =
+          new LocalActivityExecutionContext(
+              params, resultCallback, localRetryDeadline, scheduleToCloseDeadline);
+
+      LocalActivityAttemptTask task =
+          new LocalActivityAttemptTask(executionContext, params.getInitialActivityTask());
+      boolean accepted = submitAnAttempt(task);
+
+      if (accepted) {
+        if (scheduleToCloseDeadline != null) {
```

Comment:
Just `&&` ?

Reply:
I expect there may be more logic later here and it makes sense to make just an "accepted" code structure, it will be less error-prone later.